modules/tinymce/src/plugins/anchor/main/ts/core/Anchor.ts

```diff
@@ -5,37 +5,88 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
+import { Node, HTMLAnchorElement } from '@ephox/dom-globals';
+import { Arr, Option } from '@ephox/katamari';
 import Editor from 'tinymce/core/api/Editor';
-import { Element } from '@ephox/dom-globals';
+import RangeUtils from 'tinymce/core/api/dom/RangeUtils';
+import * as Settings from '../api/Settings';
 
-const isNamedAnchor = (editor: Editor, node: Element) => node.tagName === 'A' && editor.dom.getAttrib(node, 'href') === '';
+const namedAnchorSelector = 'a:not([href])';
+
+const isAnchor = (elm: Node): elm is HTMLAnchorElement => elm && elm.nodeName.toLowerCase() === 'a';
+const isNamedAnchor = (elm: Node): elm is HTMLAnchorElement => isAnchor(elm) && !elm.getAttribute('href') && (!!elm.getAttribute('id') || !!elm.getAttribute('name'));
+const isEmptyNamedAnchor = (elm: Node): elm is HTMLAnchorElement => isNamedAnchor(elm) && !elm.firstChild;
+
+const removeEmptyNamedAnchorsInSelection = (editor: Editor) => {
+  const dom = editor.dom;
+  RangeUtils(dom).walk(editor.selection.getRng(), (nodes) => {
+    Arr.each(nodes, (node) => {
```

Comment:
Again, this will cause plugin bloat (though much less than `Option`). You should probably use `Tools.each` instead.

Reply:
Yes, but it will still be much larger than `Tools.each` for plugins.

Reply:
Fixed